lib/acyclicGraph.ml

```diff
@@ -127,23 +126,23 @@ module Make (Point:Point) = struct
       n_nodes : int; n_edges : int;
       table : Index.table }
 
-  (** Used to cleanup mutable marks if a traversal function is
-      interrupted before it has the opportunity to do it itself. *)
-  let unsafe_cleanup_marks g =
-    let iter _ n = match n with
-      | Equiv _ -> ()
-      | Canonical n -> n.status <- NoMark
-    in
-    PMap.iter iter g.entries
+  module CN = struct
+    type t = canonical_node
+    let equal x y = x == y
```

Comment:
actually this is incorrect, we need to compare the `.canon`
for instance in backward_traverse we do (old code)
~~~ocaml
      x.status <- Visited;
      let gtge, x, g = get_gtge g x in
~~~
and `get_gtge` does `let x' = { x with gtge = ... } in x.status <- NoMark; ..., x', ...`
By handling the map locally we don't need to early revert the original `x` but we do need to keep the mark on the new `x`, we can do this by changing this
```suggestion
    let equal x y = x.canon == y.canon
```